### DISCUSSION TITLE
Type `longtext` not supported by PostgreSQL, upgrade.php fails, can't upgrade to 3.1.3

### DISCUSSION THREAD

**Discussion by NameDevRandom** [fudforum/FUDforum#41] [General]
Hi,

Unfortunately the upgrade script tries to `ALTER TABLE fud30_msg_store ALTER COLUMN data TYPE LONGTEXT` which fails.
Postgres doesn't have a `LONGTEXT `datatype, and `TEXT `supports unlimited length.
How can I skip or disable this needless operation?
I am still using FUDforum 3.0.5 and it really is time to upgrade...

Best regards,

**Comment by NameDevRandom** [comment]
Found a quick & dirty workaround myself:
add
`if ($tbl['name'] == $DBHOST_TBL_PREFIX .'msg_store') {
			continue;
		}`
in upgrade.php at line 901

**Comment by naudefj** [comment]
Great workaround!
Function _map_mysql_types_to_native()_ in _dbadmin.inc_ is supposed to map LONGTEXT to TEXT for PgSQL.
We will have to figure out why it's not properly converted...

**Comment by naudefj** [comment]
Fixed @ https://github.com/fudforum/FUDforum/commit/7005adc964efee0521ec22614e9bb316640f85d0
